htdocs/compta/paiement_charge.php

```diff
@@ -223,8 +223,13 @@
 	print '<tr><td class="tdtop">'.$langs->trans("RemainderToPay").'</td><td>'.price($total-$sumpaid,0,$outputlangs,1,-1,-1,$conf->currency).'</td></tr>';*/
 
 	print '<tr><td class="fieldrequired">'.$langs->trans("Date").'</td><td>';
-	$datepaye = dol_mktime(12, 0, 0, $_POST["remonth"], $_POST["reday"], $_POST["reyear"]);
-	$datepayment = empty($conf->global->MAIN_AUTOFILL_DATE) ? (empty($_POST["remonth"]) ?-1 : $datepaye) : 0;
+	if (GETPOSTISSET('remonth')) {
+		$datepayment = dol_mktime(12, 0, 0, GETPOST("remonth", 'int'), GETPOST('reday', 'int'), GETPOST('reyear', 'int'));
+	} elseif (!empty($conf->global->MAIN_AUTOFILL_DATE)) {
+		$datepayment = dol_mktime(12, 0, 0, dol_print_date(dol_now(), '%m'), dol_print_date(dol_now(), '%d'), dol_print_date(dol_now(), '%Y'));
+	} else {
+		$datepayment = '' ;
```

Comment:
So MAIN_AUTOFILL_DATE become useless ?  '' means autofill date with current date.

Reply:
Happy you'r here, I just commit a much better fix